msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -260,10 +258,16 @@ private void VerifyAuthorizationResult()
                     MsalErrorMessage.NoPromptFailedErrorMessage);
             }
 
+            if (_authorizationResult.Status == AuthorizationStatus.UserCancel)
+            {
+                throw new MsalClientException(_authorizationResult.Error,
```

Comment:
>_authorizationResult.Error [](start = 46, length = 26)

shouldn't this be the code for UserCancel ?

Reply:
not sure what you mean by "the code for UserCancel".

Reply:
right now you get the same error as before. I believe it should have the specific error code user cancelled... however if the current code calls that auth cancelled like you had it before then that is fine. IMO we should be more specific. Thus there is a difference to me in auth_cancelled and user_cancelled_auth. If however the meaning is the same, then the originally proposed error code is great.

Aside: I would generally still prefer to have the exceptions created in a factory. ensuring that the same exception type is created the same way everywhere.

---
In reply to: [199968924](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1107#discussion_r199968924) [](ancestors = 199968924)